### PR TITLE
Update GitHub workflow runners to Ubuntu 22.04

### PR DESCRIPTION
**Pull request by Nirusu**
Let's see if anything breaks.

### CODE REVIEW

**Comment by Nirusu** [comment]
grep needed some fixes. Now returns 0 even when nothing is found:
```bash
  The --files-without-match (-L) option has reverted to its behavior
  in grep 3.1 and earlier.  That is, grep -L again succeeds when a
  line is selected, not when a file is listed.  The behavior in grep
  3.2 through 3.4 was causing compatibility problems.
```
https://lists.gnu.org/archive/html/info-gnu/2020-09/msg00009.html

Love fixing compatibility problems by introducing compatibility problems :)

I rewrote the script to actually check the length of the output instead. I think that should work better.

But except from that everything seems to be working okay.

File: .github/workflows/check-licenses.yml [hunk 1]
```diff
@@ -35,4 +35,12 @@ jobs:
...
-        run: "! grep -rL --include='*.go' -e'SPDX-License-Identifier: AGPL-3.0-only' -e'DO NOT EDIT'"
+        run: |
+          FILES_WITHOUT_LICENSE_HEADER=$(grep -rL --include='*.go' -e'SPDX-License-Identifier: AGPL-3.0-only' -e'DO NOT EDIT')
+          if [[ $(echo "$FILES_WITHOUT_LICENSE_HEADER" | sed '/^\s*$/d' | wc -l) -ne "0" ]]; then
+            echo "$FILES_WITHOUT_LICENSE_HEADER"
+            exit 1
+          else
+            exit 0
+          fi
+        shell: bash
```
Comment (thomasten) [inline .github/workflows/check-licenses.yml:46]:
```suggestion
        run: "! grep -rL --include='*.go' -e'SPDX-License-Identifier: AGPL-3.0-only' -e'DO NOT EDIT' | grep ''"
```

Comment (Nirusu) [reply .github/workflows/check-licenses.yml:46]: That's definitely easier!

**Review by katexochen — APPROVED** [review]
> Let's see if anything breaks.

Looks good. :grinning:

**Pull request merged**